### ISSUE TITLE
About Parameters

### ISSUE THREAD

**Issue by neverneverendup** [jacklxc/ParagraphJointModel#5]
Dear Author,  thanks for your extradinary work!

 I want to know the exaclty epoch number of  the fever-pretrained model you used to retrain your joint model. Pretrain on FEVER is too time-cost and I don't want to train too many epoches.

BTW, how do you pretrain your pipeline model on FEVER dataset? There seems no related code in the repo.

**Comment by jacklxc** [comment]
I trained on FEVER for 3 epochs, but I don't think it's very important, as long as you complete 1 epoch of training.

As for the pipeline model, only the stance prediction modules are trained on FEVER, which is analogous to SciFact's VeriSci solution.

**Comment by neverneverendup** [comment]
OK, thank you for your reply！

When I train the joint model directly on SCIFACT without pretrain on FEVER I got nan loss on epoch 0.
The parameters are here:
```
    argparser.add_argument('--bert_lr', type=float, default=1e-5, help="Learning rate for BERT-like LM")
    argparser.add_argument('--lr', type=float, default=5e-6, help="Learning rate")
    argparser.add_argument('--dropout', type=float, default=0, help="embedding_dropout rate")
    argparser.add_argument('--bert_dim', type=int, default=1024, help="bert_dimension")
    argparser.add_argument('--epoch', type=int, default=15, help="Training epoch")
    argparser.add_argument('--MAX_SENT_LEN', type=int, default=512)
    argparser.add_argument('--loss_ratio', type=float, default=6)
  
    argparser.add_argument('--update_step', type=int, default=10)
    argparser.add_argument('--batch_size', type=int, default=1) # roberta-large: 2; bert: 8
    argparser.add_argument('--k', type=int, default=12)
    argparser.add_argument('--downsample_n', type=int, default=1)
    argparser.add_argument('--downsample_p', type=float, default=0.5)
```

nan loss:
```
Epoch 0, iter 6189, loss: 0.1849, stance loss: 0.0959, rationale loss: 0.0889:  61%|██████    | 61
Epoch 0, iter 6189, loss: 0.1849, stance loss: 0.0959, rationale loss: 0.0889:  61%|██████    | 61
Epoch 0, iter 6189, loss: 0.1849, stance loss: 0.0959, rationale loss: 0.0889:  61%|██████    | 61
Epoch 0, iter 6189, loss: 0.1849, stance loss: 0.0959, rationale loss: 0.0889:  61%|██████    | 61
Epoch 0, iter 6189, loss: 0.1849, stance loss: 0.0959, rationale loss: 0.0889:  61%|██████    | 61
Epoch 0, iter 6189, loss: 0.1849, stance loss: 0.0959, rationale loss: 0.0889:  61%|██████    | 61
Epoch 0, iter 6189, loss: 0.1849, stance loss: 0.0959, rationale loss: 0.0889:  61%|██████    | 61
Epoch 0, iter 6189, loss: 0.1849, stance loss: 0.0959, rationale loss: 0.0889:  61%|██████    | 61
Epoch 0, iter 6189, loss: 0.1849, stance loss: 0.0959, rationale loss: 0.0889:  61%|██████    | 61
Epoch 0, iter 6199, loss: nan, stance loss: nan, rationale loss: nan:  61%|██████    | 6199/10211 
Epoch 0, iter 6199, loss: nan, stance loss: nan, rationale loss: nan:  61%|██████    | 6200/10211 
Epoch 0, iter 6199, loss: nan, stance loss: nan, rationale loss: nan:  61%|██████    | 6202/10211
 Epoch 0, iter 6199, loss: nan, stance loss: nan, rationale loss: nan:  61%|██████    | 6203/10211
 Epoch 0, iter 6199, loss: nan, stance loss: nan, rationale loss: nan:  61%|██████    | 6204/10211
 Epoch 0, iter 6199, loss: nan, stance loss: nan, rationale loss: nan:  61%|██████    | 6205/10211 
Epoch 0, iter 6199, loss: nan, stance loss: nan, rationale loss: nan:  61%|██████    | 6206/10211 
Epoch 0, iter 6199, loss: nan, stance loss: nan, rationale loss: nan:  61%|██████    | 6207/10211
 Epoch 0, iter 6199, loss: nan, stance loss: nan, rationale loss: nan:  61%|██████    | 6208/10211 
Epoch 0, iter 6199, loss: nan, stance loss: nan, rationale loss: nan:  61%|██████    | 6209/10211 
```

I set the k_train= 12, and the other parameters are same with your original code in `scifact_joint_paragraph_dynamic.py` .  

When I set k_train = 0, I still get nan loss at epoch 0.   I don't know what's wrong...

**Comment by jacklxc** [comment]
Your printout looks strange, but it's hard to debug for you remotely since you cannot guarantee the problem is due to what you shared. You can start from the checkpoints I shared.

**Comment by jacklxc** [comment]
FYI the checkpoint that is pretrained on FEVER is available in README.
